thunder/dynamo/report.py

```diff
@@ -166,3 +198,235 @@ def print_sorted_memory_info(compute_t: str):
+        # way to set that with print_readable.
+        readable = _readable(self.graph, "DynamoModule", print_output=False)
+        has_cuda_args = any(hasattr(arg, "device") and arg.device.type == "cuda" for arg in inputs)
+
+        comment_str = f'''"""
+Environment information get from `torch.utils.collect_env.get_pretty_env_info()`:
+{torch_env}
+
+Versions of Thunder related libraries:
+{thunder_pkgs}"""
+'''
+        del torch_env, thunder_pkgs
+
+        code_str = ""
+        code_str += f"""# NOTE: This script requires `pytest-benchmark==4.0.0` to be installed.
+# To execute the script, run `pytest {self.graph_name}.py --benchmark-timer=torch.utils.benchmark.utils.timer.timer --benchmark-warmup=on --benchmark-group-by=param:compute_type`
+# To check the peak allocated CUDA memory, use --benchmark-json=json_file_name and look at the "max_allocated_memory_MB" field in the json file
+# To run tests for a specific compute_type, use the pytest `-k` option.
+# For example:
+#   - `-k "forward"` will run only the forward pass.
+#
+# Available options:
+#   - compute_type: "forward", "backward"
+
+import pytest
+from thunder.benchmarks.targets import parametrize_compute_type_only_training, benchmark_for_compute_type
+"""
+        # The packages that are likely to be used by the code generated from the Torch GraphModule
+        code_str += "\n".join([v.import_str for v in torch.fx.graph._custom_builtins.values()]) + "\n"
+        code_str += "\n".join(custom_import_str) + "\n"
+
+        code_str += f"""
+
+@parametrize_compute_type_only_training"""
+        func_str = f"def test_{self.graph_name}(benchmark, compute_type):\n{readable}\n"
+
+        if any(arg is None for arg in inputs):
+            func_str += f"# Warning: The inputs that cannot be inferred are set to None, requiring the user to manually give inputs according to the code\n"
+        if save_input_tensor:
+            example_inputs = eval(f"[\n{chr(10).join(arg_like(a) for a in inputs)}]")
+            input_file_name = folder / f"{self.graph_name}_inputs.pt"
+            torch.save(example_inputs, input_file_name)
+            func_str += f"inputs = torch.load('{input_file_name}')\n"
+        else:
+            input_str = f"""inputs = [\n{chr(10).join(arg_like(a) for a in inputs)}\n"""
+            func_str += f"{_addindent(input_str, 4)}\n]\n"
+            del input_str
+
+        func_str = f"""{func_str}
+model = DynamoModule()
+{custom_executor_str}
+"""
+        if not has_cuda_args:
+            func_str += f"""benchmark(compiled, *inputs)"""
+        else:
+            func_str += f"""benchmark_for_compute_type(compute_type, benchmark, compiled_model, inputs, {{}})"""
+
+        with open(folder / f"{self.graph_name}_{custom_executor_name}.py", "w") as f:
+            print(comment_str, file=f)
+            print(code_str, file=f)
+            print(_addindent(func_str, 4), file=f)
+
+
+class FXGraphReport(FXGraphReportBase):
```

Comment:
I wouldn't make these subclasses of `FXGraphReportBase`. Because they don't have additional data, I think all three classes can just be `FXGraphReport` and then `write_repro` can become `write_benchmark` or `write_repro` and its arguments can change